nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,866 @@
+
+        `std_coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the standard deviation of the models weight vector across
+            fold for each class.
+
+        `std_coef_img_`: dict of Nifti1Image
+            Dictionary containing `std_coef_` with class names as keys,
+            and `coef_` transformed in Nifti1Image as values. In the case
+            of a regression, it contains a single Nifti1Image at the key 'beta'.
+
+        `cv_y_true_`: numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Ground truth labels for left out samples in inner cross-validation.
+
+        `cv_y_pred_`: numpy.ndarray, shape=(n_samples * n_folds, n_classes)
+            Predicted labels for left out samples in inner cross-validation.
+
+        `cv_params_`: dict of lists
+            Best point in the parameter grid for each tested fold
+            in the inner cross validation loop.
+
+        `cv_indices_`: numpy.ndarray, shape=(n_samples * n_folds)
+            Indices of the inner cross-validation folds.
+
+        `cv_scores_`: dict, (classes, n_folds)
+            Scores (misclassification) for each parameter, and on each fold
+        """
+        self._check_estimator()
+        self.memory_ = _check_memory(self.memory, self.verbose)
+    
+        # Nifti masking
+        self.masker_ = check_embedded_nifti_masker(self, multi_subject=False)
+        X = self.masker_.fit_transform(X)
+        self.mask_img_ = self.masker_.mask_img_
+
+        X, y = check_X_y(X, y, dtype=np.float, multi_output=True)
+
+        # Setup scorer
+        scorer = check_scoring(self.estimator, self.scoring)
+        self.cv_ = list(check_cv(
+            self.cv, y=y, classifier=self.is_classif).split(X, y,
+                                                            groups=groups))
+
+        # Define the number problems to solve. In case of classification this
+        # number corresponds to the number of binary problems to solve
+        if self.is_classif:
+            y = self._binarize_y(y)
+        else:
+            y = y[:, np.newaxis]
+        if self.is_classif and self.n_classes_ > 2:
+            n_problems = self.n_classes_
+        else:
+            n_problems = 1
+
+        # Return a suitable screening percentile according to the mask image
+        self.screening_percentile_ = _adjust_screening_percentile(
+            self.screening_percentile, self.mask_img_, verbose=self.verbose)
+
+        coefs = {}
+        intercepts = {}
+        cv_y_prob = {}
+        cv_y_true = {}
+        cv_indices = {}
+        cv_scores = {}
+        self.cv_params_ = {}
+        classes = self.classes_
+
+        parallel = Parallel(n_jobs=self.n_jobs, verbose=2 * self.verbose)
+
+        for i, (class_index, coef, intercept, y_info, params, scores) \
+                in enumerate(parallel(delayed(self._cache(_parallel_fit))(
+                    self.estimator, X, y[:, c], train, test,
```

Comment:
it would probably be faster to move the iteration over classes to the innermost loop inside `_parallel_fit` so that estimators that naturally handle multiple outputs like ridge can regress all dependent variables at once

Reply:
I believe this has been addressed or will be addressed ina new PR?

Reply:
This has been addressed in our working session together IRL. We decided to keep iteration over classes because of `screening_percentile_` cannot work with multi-class classification, and also to avoid complication when compiling results of `parallel_fit`.